lib/fluent/diagtool/diagutils.rb

```diff
@@ -139,19 +147,23 @@ def run_diagtool()
       tdconf = c.collect_tdconf()
       diaglogger_info("[Collect] config file is stored in #{tdconf}")
 
-      diaglogger_info("[Collect] Collecting #{@conf[:package_name]} gem information...")
-      tdgem = c.collect_tdgems()
-      diaglogger_info("[Collect] #{@conf[:package_name]} gem information is stored in #{tdgem}")
-
-      gem_info = c.collect_manually_installed_gems(tdgem)
-      diaglogger_info("[Collect] #{@conf[:package_name]} gem information (bundled by default) is stored in #{gem_info[:bundled]}")
-      diaglogger_info("[Collect] #{@conf[:package_name]} manually installed gem information is stored in #{gem_info[:local]}")
-      local_gems = File.read(gem_info[:local]).lines(chomp: true)
-      unless local_gems == [""]
-        diaglogger_info("[Collect] #{@conf[:package_name]} manually installed gems:")
-        local_gems.each do |gem|
-          diaglogger_info("[Collect]   * #{gem}")
+      case @type
```

Comment:
you can't use @type in this context. 
```suggestion
      case @conf[:type]
```